cloudsync/sync/state.py

```diff
@@ -1014,7 +1014,7 @@ def update(self, side, otype, oid, path=None, hash=None, exists=True, prior_oid=
                 ent.ignored = IgnoreReason.NONE
 
             if prior_ent and not prior_ent.is_discarded:
-                if not ent or not ent.is_conflicted:
+                if not ent or (not ent.is_conflicted and (prior_ent[side].sync_hash or not ent[side].sync_hash)):
```

Comment:
Can we add a comment here that explains the logic?

Reply:
Well, I'm not fully aware of the purpose of the "not ent.is_conflicted" logic, but I can try to explain the rest...

Reply:
Ya I think just anything that would give the next person that looks at this code block a head start on making sense of the logic would be useful

Reply:
I added the comment, you can check it out

Reply:
Looks good